autogen/agentchat/contrib/capabilities/transforms.py

```diff
@@ -200,6 +202,113 @@ def _validate_max_tokens(self, max_tokens: Optional[int] = None) -> Optional[int
         return max_tokens if max_tokens is not None else sys.maxsize
 
 
+class TextMessageCompressor:
+    """A transform for compressing text messages in a conversation history.
+
+    It uses a specified text compression method to reduce the token count of messages, which can lead to more efficient
+    processing and response generation by downstream models.
+    """
+
+    def __init__(
+        self, text_compressor: Optional[TextCompressor] = None, messages_to_compress: Literal["last", "all"] = "last"
+    ):
```

Comment:
Maybe the option "all" is a bit too dangerous, especially as a constructor argument that is accessed by the user. Since, it could lead to considerable delays and possible information loss.

Reply:
You raise a good point. My aim was to address scenarios like an agent joining a group chat midway, where it's crucial for the agent to grasp the full conversation context. Compressing only the latest message might not be sufficient. `TransformMessages` capability hooks to `process_all_messages_before_reply`. 
This ensures that information loss is minimized and mainly impacts agents that have this transform applied (or any other transform like token truncation). Let me know what you think.

Reply:
In terms of performance, I'm letting the user decide what trade-offs to make

Reply:
Not certain if this is an optimal solution for several reasons:

* This specific initialization might also be necessary for future transformations, not just limited to 'TextMessageCompressor'.
* With the current implementation, to apply the 'TextMessageCompressor' transformation to a newly created agent, you need to instantiate two 'TextMessageCompressor' objects: one to apply the transformation to all available messages and another to apply it to each new message. This implementation could be improved.
* Opting for 'init_apply_transform', it becomes effortless to simply 'pass' it for transformations it doesn't apply.

Reply:
Simple and effective implementation indeed! I would only suggest a few changes in fields' names, to make it clearer.